### ISSUE TITLE
KeyError when using SlimPruner with AGPPruner

### ISSUE THREAD

**Issue by crawlingcub** [microsoft/nni#3900]
**Describe the issue**:

Hi, 

The Slim pruner expects a `sparsity` key in the pruner configuration. However, AGP's config contains `initial_sparsity` and `final_sparsity`. Should the initialization logic in `SlimPrunerMasker` be moved to `calc_mask` instead?

I will be happy to raise a PR based on your suggestions.

Let me know if you need more info.

Thanks!

**Environment**:
- NNI version: 2.2
- Training service (local|remote|pai|aml|etc): local
- Client OS: ubuntu
- Server OS (for remote mode only):
- Python version: 3.7
- PyTorch/TensorFlow version: Pytorch 1.8
- Is conda/virtualenv/venv used?: conda
- Is running in Docker?: no

**Log message**:
 ```python
Traceback (most recent call last):
...
  File "/home/ubuntu/anaconda3/envs/pytorch_latest_p37/lib/python3.7/site-packages/nni/algorithms/compression/pytorch/pruning/agp.py", line 46, in __init__
    self.masker = MASKER_DICT[pruning_algorithm](model, self)
  File "/home/ubuntu/anaconda3/envs/pytorch_latest_p37/lib/python3.7/site-packages/nni/algorithms/compression/pytorch/pruning/structured_pruning.py", line 684, in __init__
    k = int(all_bn_weights.shape[0] * pruner.config_list[0]['sparsity'])
KeyError: 'sparsity'
```

**How to reproduce it?**:

Running AGP with `pruning_algorithm="slim"` can reproduce this error.

**Comment by J-shang** [comment]
Hi @crawlingcub, we fix this in `nni v2.3`, you could try agp with slim in the new version.

**Comment by crawlingcub** [comment]
Thanks, I will try it out

**Comment by scarlett2018** [comment]
> Hi @crawlingcub, we fix this in `nni v2.3`, you could try agp with slim in the new version.

closing it as it has been fixed in v2.3. @crawlingcub feel free to reopen if the fix does not work well for you, thanks.
